posthog/hogql/resolver.py

```diff
@@ -494,6 +525,29 @@ def visit_compare_operation(self, node: ast.CompareOperation):
         node.type = ast.BooleanType()
         return node
 
+    def _is_events_table(self, node: ast.Expr) -> bool:
+        if isinstance(node, ast.Field):
+            try:
+                return isinstance(node.type.table_type.table, EventsTable)
+            except:
+                return False
+        return False
+
+    def _is_s3_cluster(self, node: ast.Expr) -> bool:
+        if isinstance(node, ast.SelectQuery):
+            try:
+                return isinstance(node.select_from.type.table_type.table, S3Table)
+            except:
```

Comment:
> main thing I was trying to protect here was if the ast access path changed (due to implementation change or something else)

Hmm, if there's an implementation change though, we'll have a silently non-functioning mechanism, which seems dangerous. Considering our test coverage of HogQL, I would be more comfortable with this failing loudly